pkg/appauth/manager/json/json.go

```diff
@@ -17,3 +17,185 @@
 // or submit itself to any jurisdiction.
 
 package json
+
+import (
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"reflect"
+	"time"
+
+	apppb "github.com/cs3org/go-cs3apis/cs3/auth/applications/v1beta1"
+	authpb "github.com/cs3org/go-cs3apis/cs3/auth/provider/v1beta1"
+	userpb "github.com/cs3org/go-cs3apis/cs3/identity/user/v1beta1"
+	typespb "github.com/cs3org/go-cs3apis/cs3/types/v1beta1"
+	"github.com/cs3org/reva/pkg/appauth"
+	"github.com/cs3org/reva/pkg/appauth/manager/registry"
+	"github.com/cs3org/reva/pkg/errtypes"
+	"github.com/cs3org/reva/pkg/user"
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+	"github.com/sethvargo/go-password/password"
+)
+
+func init() {
+	registry.Register("json", New)
+}
+
+type config struct {
+	File          string `mapstructure:"file"`
+	TokenStrength int    `mapstructure:"token_strength"`
+}
+
+type jsonManager struct {
+	config *config
+	model  *jsonModel
+}
+
+type jsonModel struct {
+	file  string
+	State map[string]*apppb.AppPassword `json:"state"`
+}
+
+// New returns a new mgr.
+func New(m map[string]interface{}) (appauth.Manager, error) {
+	c, err := parseConfig(m)
+	if err != nil {
+		return nil, errors.Wrap(err, "error creating a new manager")
+	}
+
+	c.init()
+
+	// load or create file
+	model, err := loadOrCreate(c.File)
+	if err != nil {
+		return nil, errtypes.InternalError("error loading the file containing the application passwords")
+	}
+
+	return &jsonManager{config: c, model: model}, nil
+}
+
+func (c *config) init() {
+	if c.File == "" {
+		c.File = "/etc/revad/appauth.json"
```

Comment:
Change this to `/var/tmp/reva/appauth.json`. We read configs from `/etc/revad` and store data in `/var/tmp`